### ISSUE TITLE
Brave News ads should not send confirmations for conversions if users have not opted-in to Brave Ads

### ISSUE THREAD

**Issue by tmancey** [brave/brave-browser#16575]
## Description 

Brave News ads should not send confirmations for conversions

## Steps to Reproduce

   1. Trigger a Brave News Ad conversion (do not enable Brave Ads)

## Actual result:

Confirmation is sent

## Expected result:
Confirmation should not be sent

## Reproduces how often: 

Easily reproduced

## Version/Channel Information:

- Can you reproduce this issue with the current release? No
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? N/A
- Does the issue resolve itself when disabling Brave Rewards? N/A
- Is the issue reproducible on the latest version of Chrome? N/A

## Miscellaneous Information:

This issue does not affect end users as Brave News ads are not currently shown to users

**Comment by stephendonner** [comment]
Verified `FIXED` using 


Brave | 1.28.88 Chromium: 92.0.4515.107 (Official Build) beta (x86_64)
-- | --
Revision | 87a818b10553a07434ea9e2b6dccf3cbe7895134-refs/branch-heads/4515@{#1634}
OS | macOS Version 11.4 (Build 20F71)

Steps:
1. new profile
2. launch Brave
3. opened a new-tab page
4. clicked on `Show Brave News`
5. scrolled down to an inline-content ad
6. checked logs


Confirmed I was given a confirmation, but not a conversion into tokens; the last log output for it is:

```
[51575:259:0722/153900.876762:VERBOSE6:inline_content_ad_event_clicked.cc(32)] Successfully logged inline content ad clicked event
[51575:259:0722/153901.083180:VERBOSE1:redeem_unblinded_token.cc(82)] OnCreateConfirmation
[51575:259:0722/153901.083264:VERBOSE6:redeem_unblinded_token.cc(84)] URL Response:
  URL: https://ads-serve.bravesoftware.com/v1/confirmation/131fcabc-4046-4d10-9ca8-18dc16c96e1c
  Response Status Code: 418
  Response: {"id":"131fcabc-4046-4d10-9ca8-18dc16c96e1c","payload":{},"createdAt":"2021-07-22T22:39:01.044Z","type":"click","modifiedAt":"2021-07-22T22:39:01.044Z","creativeInstanceId":"033a95ca-963b-460c-b203-8181346bd41c"}
[51575:259:0722/153901.083334:VERBOSE1:confirmations.cc(234)] Successfully sent confirmation with id 131fcabc-4046-4d10-9ca8-18dc16c96e1c, creative instance id 033a95ca-963b-460c-b203-8181346bd41c and click
```

I then re-ran the above, but did, instead:

1. new profile
2. launch Brave
3. opened a new-tab page
4. clicked on `Start using Rewards`
5. clicked on `Show Brave News`
6. scrolled down to an inline-content ad
7. checked logs

Now confirmed that when `Ads/Rewards` **IS** enabled, I saw confirmations **and** conversions w/tokens, etc.:

```
[51811:259:0722/154134.694826:VERBOSE6:inline_content_ad_event_clicked.cc(32)] Successfully logged inline content ad clicked event
[51811:259:0722/154134.747298:VERBOSE1:ads_impl.cc(163)] Visited URL is not supported
[51811:259:0722/154134.748276:VERBOSE1:ads_impl.cc(185)] Visited URL is not supported
[51811:259:0722/154134.748365:VERBOSE1:redeem_unblinded_token.cc(82)] OnCreateConfirmation
[51811:259:0722/154134.748416:VERBOSE6:redeem_unblinded_token.cc(84)] URL Response:
  URL: https://ads-serve.bravesoftware.com/v1/confirmation/7bb9520c-01b5-4189-8367-3da846c8570d
  Response Status Code: 418
  Response: {"id":"7bb9520c-01b5-4189-8367-3da846c8570d","payload":{},"createdAt":"2021-07-22T22:41:34.692Z","type":"click","modifiedAt":"2021-07-22T22:41:34.692Z","creativeInstanceId":"033a95ca-963b-460c-b203-8181346bd41c"}
[51811:259:0722/154134.748518:VERBOSE1:confirmations.cc(234)] Successfully sent confirmation with id 7bb9520c-01b5-4189-8367-3da846c8570d, creative instance id 033a95ca-963b-460c-b203-8181346bd41c and click
[51811:259:0722/154145.921011:VERBOSE6:ad_transfer.cc(138)] Successfully logged transferred ad event
[51811:259:0722/154146.285317:VERBOSE1:redeem_unblinded_token.cc(82)] OnCreateConfirmation
[51811:259:0722/154146.285432:VERBOSE6:redeem_unblinded_token.cc(84)] URL Response:
  URL: https://ads-serve.bravesoftware.com/v1/confirmation/c........
  Response: {"id":"c98fbf35-9ce1-492a-9c8c-c25088145fe3","payload":{},"createdAt":"2021-07-22T22:41:46.071Z","type":"landed","modifiedAt":"2021-07-22T22:41:46.071Z","creativeInstanceId":"033a95ca-963b-460c-b203-8181346bd41c"}
[51811:259:0722/154146.285734:VERBOSE1:redeem_unblinded_token.cc(112)] FetchPaymentToken
[51811:259:0722/154146.285804:VERBOSE2:redeem_unblinded_token.cc(113)] GET /v1/confirmation/{confirmation_id}/paymentToken
[51811:259:0722/154146.285858:VERBOSE5:redeem_unblinded_token.cc(117)] URL Request:
  URL: https://ads-serve.bravesoftware.com/v1/confirmation/c98fbf35-9ce1-492a-9c8c-c25088145fe3/paymentToken
  Method: GET
[51811:259:0722/154146.366649:VERBOSE1:redeem_unblinded_token.cc(128)] OnFetchPaymentToken
[51811:259:0722/154146.366767:VERBOSE6:redeem_unblinded_token.cc(130)] URL Response:
  URL: https://ads-serve.bravesoftware.com/v1/confirmation/c98fbf35-9ce1-492a-9c8c-c25088145fe3/paymentToken
  Response Status Code: 200
  Response: {"id":"c98fbf35-9ce1-492a-9c8c-c25088145fe3","createdAt":"2021-07-22T22:41:46.071Z","type":"landed","modifiedAt":"2021-07-22T22:41:46.165Z","creativeInstanceId":"033a95ca-963b-460c-b203-8181346bd41c","paymentToken":{"publicKey":"lLO5tErGoTK0askrALab6pKGAnBHqELSyw/evqZRwH8=","batchProof":"nd4GeNGHXZ6/GEebCmZFfVe66L3UuxVKCxrhwJK8SgWCOegJvg2YQ6TKpJfMg/SL3TcsV+JyBDEOk3EeFV9DDQ==","signedTokens":["TnLHEMY/YxXZijjCY7XVuSHX4/2kzYEQDHazmq8d8TI="]}}
[51811:259:0722/154146.367653:VERBOSE1:confirmations.cc(250)] Successfully redeemed unblinded token with confirmation id c98fbf35-9ce1-492a-9c8c-c25088145fe3, creative instance id 033a95ca-963b-460c-b203-8181346bd41c and landed
[51811:259:0722/154146.368287:VERBOSE1:confirmations.cc(273)] Added 1 unblinded payment token with an estimated redemption value of 0 BAT, you now have 1 unblinded payment tokens
```

**Comment by btlechowski** [comment]
`QA/Blocked` till https://github.com/brave/brave-browser/issues/17330 is fixed and uplifted to 1.18.x

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.28.104 Chromium: 92.0.4515.131 (Official Build) (64-bit)
-- | --
Revision | 6b8d6c56ce21e38a72f7c4becb5abc1fa5134f29-refs/branch-heads/4515@{#1933}
OS | Ubuntu 18.04 LTS

Verified test plan from the description

Verified conversion was triggered. 
```
[16388:16388:0809/210154.207093:VERBOSE1:conversions.cc(322)] Conversion for campaign id 28dbd338-aaf8-44c0-b3d9-01df58ce2c04, creative set id 1ab4f8ae-4a3e-47a7-a6fe-9ab662ad17cf, creative instance id e837131f-13da-46ae-b8d2-a7f9d00efb2f and advertiser id 401acfad-2cbb-4bfc-b522-ceae781f3045 on Monday, August 9, 2021 at 9:01:54 PM
[16388:16388:0809/210154.240293:VERBOSE6:conversions.cc(375)] Successfully logged conversion event
[16388:16388:0809/210154.240385:VERBOSE3:conversions.cc(398)] Successfully appended conversion to queue
[16388:16388:0809/210154.243043:VERBOSE1:conversions.cc(503)] Convert campaign id 28dbd338-aaf8-44c0-b3d9-01df58ce2c04, creative set id 1ab4f8ae-4a3e-47a7-a6fe-9ab662ad17cf, creative instance id e837131f-13da-46ae-b8d2-a7f9d00efb2f and advertiser id 401acfad-2cbb-4bfc-b522-ceae781f3045 in 7 hours, 12 minutes, 29 seconds at 4:14 AM
```

Verified confirmation is not sent for conversion
```
[17603:17603:0810/210418.596613:VERBOSE1:conversions.cc(444)] Successfully converted ad with campaign id 28dbd338-aaf8-44c0-b3d9-01df58ce2c04, creative set id 1ab4f8ae-4a3e-47a7-a6fe-9ab662ad17cf, creative instance id e837131f-13da-46ae-b8d2-a7f9d00efb2f and advertiser id 401acfad-2cbb-4bfc-b522-ceae781f3045 on Tuesday, August 10, 2021 at 4:14:23 AM
[17603:17603:0810/210418.618663:VERBOSE3:conversions.cc(413)] Successfully removed conversion from queue
```
